verifier/src/x509/rfc4518.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! An implementation of https://www.rfc-editor.org/rfc/rfc4518 string preparation.
+//!
+//! A good document on normal forms, https://unicode.org/reports/tr15/#Norm_Forms
+extern crate alloc;
+
+use alloc::string::String;
+use unicode_normalization::UnicodeNormalization;
+
+/// An RFC4518 prepared String
+#[derive(Debug, PartialEq)]
+pub struct Rfc4518String {
+    inner: String,
+}
+
+impl From<&str> for Rfc4518String {
+    fn from(string: &str) -> Self {
+        let normalized = string
+            .chars()
+            .filter_map(rfc_4518_filter_map)
+            .nfkc()
+            .collect::<String>();
```

Comment:
I added a task to #50 for the missing bidi and prohibited unicode code points logic